pkg/logging/logging.go

```diff
@@ -0,0 +1,88 @@
+package logging
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type LogrusFileHook struct {
+	file      *os.File
+	flag      int
+	chmod     os.FileMode
+	formatter *logrus.TextFormatter
+}
+
+var Debug bool
+
+func NewLogrusFileHook(file string, flag int, chmod os.FileMode) (*LogrusFileHook, error) {
+	plainFormatter := &logrus.TextFormatter{DisableColors: true}
+	logFile, err := os.OpenFile(file, flag, chmod)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to write file on filehook %v", err)
+		return nil, err
+	}
+
+	return &LogrusFileHook{logFile, flag, chmod, plainFormatter}, err
+}
+
+func (hook *LogrusFileHook) Fire(entry *logrus.Entry) error {
+
+	plainformat, err := hook.formatter.Format(entry)
+	line := string(plainformat)
+	_, err = hook.file.WriteString(line)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to write file on filehook(entry.String)%v", err)
+		return err
+	}
+
+	return nil
+}
+
+func (hook *LogrusFileHook) Levels() []logrus.Level {
+	return logrus.AllLevels
+}
+
+type StandardHook struct {
+	Writer    io.Writer
+	LogLevels []logrus.Level
+	Formatter *logrus.TextFormatter
+}
+
+func (hook *StandardHook) Fire(entry *logrus.Entry) error {
+	text, err := hook.Formatter.Format(entry)
+	if err != nil {
+		return err
+	}
+
+	if !Debug && entry.Level == logrus.InfoLevel || entry.Level == logrus.DebugLevel {
```

Comment:
I don't fully understand this `if` here. Can you elaborate a little bit longer ? I mean, how these are evaluated ?

Reply:
as I understand it, it should work like ( if debug is true ) and ( line loglevel == info or debug )